java/algorithm/src/main/java/org/apache/arrow/algorithm/sort/DefaultVectorComparators.java

```diff
@@ -56,6 +61,14 @@
         return (VectorValueComparator<T>) new Float4Comparator();
       } else if (vector instanceof Float8Vector) {
         return (VectorValueComparator<T>) new Float8Comparator();
+      } else if (vector instanceof UInt1Vector) {
+        return (VectorValueComparator<T>) new UInt1Comparator();
+      } else if (vector instanceof UInt2Vector) {
+        return (VectorValueComparator<T>) new UInt2Comparator();
+      } else if (vector instanceof UInt4Vector) {
+        return (VectorValueComparator<T>) new UInt4Comparator();
+      } else if (vector instanceof UInt8Vector) {
+        return (VectorValueComparator<T>) new UInt8Comparator();
```

Comment:
easier to read/more performant if we have map of class names to comparators?

Reply:
@praveenbingo Thanks a lot for your comments.
I agree with you that using a map with vector class as key is more performant. 

The reason that we use the vector to create the comparator is that, for some cases, the comparator cannot be created without the full knowledge of the vector.
For example, for a ListVector, we cannot create a comparator for it, unless we have information about its inner vector.